In [1]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))



# Overview
**These are the questions I will answer through this review.**

1) The 10 highest scorers of the decade of the season

2) Top 10 scorers of all seasons with more than 35 points per game.

3) 1st year statistics of each player.

4) 2nd year stats for each player.

5) 5 years statistics of each player.

6) Curiosities / Relevant data


# Visão geral
**Estas são as perguntas que vou responder através desta análise.**

1) Os 10 maiores pontuadores por década de temporada

2) Os 10 maiores pontuadores de todas temporadas com mais de 35 pontos por partida.

3) Estatísticas do 1º ano de cada jogador.

4) Estatísticas do 2º ano de cada jogador.

5) Estatísticas dos 5 anos de cada jogador.

6) Curiosidades / Dados relevantes



In [2]:
player_game = pd.read_csv("input/nba-aba-baa-stats/Player Per Game.csv")
player_by_play = pd.read_csv("input/nba-aba-baa-stats/Player Play By Play.csv")
allstar = pd.read_csv("input/allstar-games-stats/allstar_games_stats.csv")


In [3]:
player_game.dtypes

seas_id            int64
season             int64
player_id          int64
player            object
birth_year       float64
pos               object
age              float64
experience         int64
lg                object
tm                object
g                  int64
gs               float64
mp_per_game      float64
fg_per_game      float64
fga_per_game     float64
fg_percent       float64
x3p_per_game     float64
x3pa_per_game    float64
x3p_percent      float64
x2p_per_game     float64
x2pa_per_game    float64
x2p_percent      float64
e_fg_percent     float64
ft_per_game      float64
fta_per_game     float64
ft_percent       float64
orb_per_game     float64
drb_per_game     float64
trb_per_game     float64
ast_per_game     float64
stl_per_game     float64
blk_per_game     float64
tov_per_game     float64
pf_per_game      float64
pts_per_game     float64
dtype: object

In [4]:
player_by_play.dtypes

seas_id                               int64
season                                int64
player_id                             int64
player                               object
birth_year                          float64
pos                                  object
age                                   int64
experience                            int64
lg                                   object
tm                                   object
g                                     int64
mp                                    int64
pg_percent                          float64
sg_percent                          float64
sf_percent                          float64
pf_percent                          float64
c_percent                           float64
on_court_plus_minus_per_100_poss    float64
net_plus_minus_per_100_poss         float64
bad_pass_turnover                     int64
lost_ball_turnover                    int64
shooting_foul_committed               int64
offensive_foul_committed        

Retornar o primeiro ano de cada jogador

In [5]:
#player_game.query("player =  'players' and experience == 2")
player_game.query('player in ["Michael Jordan", "Kobe Bryant"] and experience == 1')


#print(df.query('state in ["NY", "TX"]'))
#'state == "CA"'


seas_id  season  player_id          player  birth_year pos   age  \
15809    15419    1997       3000     Kobe Bryant         NaN  SF  18.0   
21195    10057    1985       2193  Michael Jordan         NaN  SG  21.0   

       experience   lg   tm  ...  ft_percent  orb_per_game  drb_per_game  \
15809           1  NBA  LAL  ...       0.819           0.7           1.2   
21195           1  NBA  CHI  ...       0.845           2.0           4.5   

       trb_per_game  ast_per_game  stl_per_game  blk_per_game  tov_per_game  \
15809           1.9           1.3           0.7           0.3           1.6   
21195           6.5           5.9           2.4           0.8           3.5   

       pf_per_game  pts_per_game  
15809          1.4           7.6  
21195          3.5          28.2  

[2 rows x 35 columns]

In [6]:
allstar.query("Player == 'Lebron James' and Age == 20")

Season  Age   Tm   Lg Pos    G   GS     MP   FG   FGA  ...  FT%  ORB  TRB  \
0  2004-05   20  CLE  NBA  SF  1.0  1.0  31:25  6.0  13.0  ...  0.0  1.0  8.0   

   AST  STL  BLK  TOV   PF   PTS        Player  
0  6.0  2.0  0.0  3.0  0.0  13.0  Lebron James  

[1 rows x 26 columns]

In [7]:
#Retorna todos jogadores no primeira temporada
jogadores_primeiro_ano = player_game.query("experience == 1 and pts_per_game > 20")


maiores_pontuadores = jogadores_primeiro_ano.sort_values(by=['pts_per_game'], ascending=False)
print (maiores_pontuadores[['player','age','season', 'experience','pts_per_game']])

#print(df[['name', 'score']])
#['player','age','season', 'experience','pts_per_game']
       

                 player   age  season  experience  pts_per_game
29044  Wilt Chamberlain  23.0    1960           1          37.6
28806      Walt Bellamy  22.0    1962           1          31.6
28888   Oscar Robertson  22.0    1961           1          30.5
27170   Spencer Haywood  20.0    1970           1          30.0
26472         Dan Issel  22.0    1971           1          29.9
...                 ...   ...     ...         ...           ...
9873       Kevin Durant  19.0    2008           1          20.3
4491        Joel Embiid  22.0    2017           1          20.2
21719     Clark Kellogg  21.0    1983           1          20.1
14103       Elton Brand  20.0    2000           1          20.1
8959       Tyreke Evans  20.0    2010           1          20.1

[70 rows x 5 columns]


# Retornar a primeira e a última  temporada do dataset

In [8]:
primeira_temporada = player_game['season'].min()
ultima_temporada = player_game['season'].max()
   

print('A primeira temporada é: ',primeira_temporada, 'e a última temporada é:' ,ultima_temporada)


A primeira temporada é:  1947 e a última temporada é: 2023


# Definir os marcos das temporadas por década para utilizar na próxima análise


In [9]:
temp_40 = 1947
temp_50 = 1950
temp_60 = 1960
temp_70 = 1970
temp_80 = 1980
temp_90 = 1990 
temp_2000 = 2000
temp_2010 = 2010
temp_2020 = 2020
temp_ultima = ultima_temporada

#  Verificar a média de pontos por jogo  dos 10 maiores pontuadores para utilizar na próxima análise

# Década de 40

In [10]:
periodo_inicial = temp_40
periodo_final = temp_50

pontuadores = player_game[(player_game['season'] >= periodo_inicial) & (player_game['season'] <= periodo_final)]
decada = pontuadores.nlargest(10, 'pts_per_game')[['player','pos','age', 'season', 'experience','tm','pts_per_game']]
media_decada = decada["pts_per_game"].mean()

print('A média da década', periodo_inicial , 'foi de: ', media_decada.round(2))
print('Os maiores pontuadores da década ', periodo_inicial )
print(decada)

A média da década 1947 foi de:  23.05
Os maiores pontuadores da década  1947
              player pos   age  season  experience   tm  pts_per_game
30661   George Mikan   C  24.0    1949           1  MNL          28.3
30379   George Mikan   C  25.0    1950           2  MNL          27.4
30708      Joe Fulks  PF  27.0    1949           3  PHW          26.0
30246     Alex Groza   C  23.0    1950           1  INO          23.4
31049      Joe Fulks  PF  25.0    1947           1  PHW          23.2
30878      Joe Fulks  PF  26.0    1948           2  PHW          22.1
30900  Max Zaslofsky  SG  22.0    1948           2  CHS          21.0
30742  Max Zaslofsky  SG  23.0    1949           3  CHS          20.6
30841    Ed Sadowski   C  30.0    1948           2  BOS          19.4
30997    Ed Sadowski   C  29.0    1947           1  TRH          19.1


# Década de 50

In [11]:
periodo_inicial = temp_50
periodo_final = temp_60

pontuadores = player_game[(player_game['season'] >= periodo_inicial) & (player_game['season'] <= periodo_final)]
decada = pontuadores.nlargest(10, 'pts_per_game')[['player','pos','age', 'season', 'experience','tm','pts_per_game']]
media_decada = decada["pts_per_game"].mean()

print('A média da década', periodo_inicial , 'foi de: ', media_decada.round(2))
print('Os maiores pontuadores da década ', periodo_inicial )
print(decada)

A média da década 1950 foi de:  28.95
Os maiores pontuadores da década  1950
                 player pos   age  season  experience   tm  pts_per_game
29044  Wilt Chamberlain   C  23.0    1960           1  PHW          37.6
28991       Jack Twyman  SF  25.0    1960           5  CIN          31.2
28971      Elgin Baylor  SF  25.0    1960           2  MNL          29.6
29057        Bob Pettit  PF  26.0    1959           5  STL          29.2
30152      George Mikan   C  26.0    1951           3  MNL          28.4
29205    George Yardley  SF  29.0    1958           5  DET          27.8
30379      George Mikan   C  25.0    1950           2  MNL          27.4
29124       Paul Arizin  SF  30.0    1959           7  PHW          26.4
28939        Bob Pettit  PF  27.0    1960           6  STL          26.1
29104       Jack Twyman  SF  24.0    1959           4  CIN          25.8


# Década de 60

In [12]:
periodo_inicial = temp_60
periodo_final = temp_70

pontuadores = player_game[(player_game['season'] >= periodo_inicial) & (player_game['season'] <= periodo_final)]
decada = pontuadores.nlargest(10, 'pts_per_game')[['player','pos','age', 'season', 'experience','tm','pts_per_game']]
media_decada = decada["pts_per_game"].mean()

print('A média da década', periodo_inicial , 'foi de: ', media_decada.round(2))
print('Os maiores pontuadores da década ', periodo_inicial )
print(decada)

A média da década 1960 foi de:  39.04
Os maiores pontuadores da década  1960
                 player pos   age  season  experience   tm  pts_per_game
28813  Wilt Chamberlain   C  25.0    1962           3  PHW          50.4
28679  Wilt Chamberlain   C  26.0    1963           4  SFW          44.8
28413  Wilt Chamberlain   C  28.0    1965           6  SFW          38.9
28921  Wilt Chamberlain   C  24.0    1961           2  PHW          38.4
28734      Elgin Baylor  SF  27.0    1962           4  LAL          38.3
29044  Wilt Chamberlain   C  23.0    1960           1  PHW          37.6
28549  Wilt Chamberlain   C  27.0    1964           5  SFW          36.9
28124        Rick Barry  SF  22.0    1967           2  SFW          35.6
28850      Elgin Baylor  SF  26.0    1961           3  LAL          34.8
28412  Wilt Chamberlain   C  28.0    1965           6  TOT          34.7


# Década de 70

In [13]:
periodo_inicial = temp_70
periodo_final = temp_80

pontuadores = player_game[(player_game['season'] >= periodo_inicial) & (player_game['season'] <= periodo_final)]
decada = pontuadores.nlargest(10, 'pts_per_game')[['player','pos','age', 'season', 'experience','tm','pts_per_game']]
media_decada = decada["pts_per_game"].mean()

print('A média da década', periodo_inicial , 'foi de: ', media_decada.round(2))
print('Os maiores pontuadores da década ', periodo_inicial )
print(decada)

A média da década 1970 foi de:  32.84
Os maiores pontuadores da década  1970
                    player pos   age  season  experience   tm  pts_per_game
26198  Kareem Abdul-Jabbar   C  24.0    1972           3  MIL          34.8
25981        Charlie Scott  SG  23.0    1972           2  VIR          34.6
24690           Bob McAdoo   C  23.0    1975           3  BUF          34.5
25876       Tiny Archibald  PG  24.0    1973           3  KCO          34.0
22902        George Gervin  SG  27.0    1980           8  SAS          33.1
25745        Julius Erving  SF  22.0    1973           2  VIR          31.9
26643  Kareem Abdul-Jabbar   C  23.0    1971           2  MIL          31.7
26297           Rick Barry  SF  27.0    1972           6  NYA          31.5
27032           Jerry West  PG  31.0    1970          10  LAL          31.2
24143        Pete Maravich  SG  29.0    1977           7  NOJ          31.1


# Década de 80

In [14]:
periodo_inicial = temp_80
periodo_final = temp_90

pontuadores = player_game[(player_game['season'] >= periodo_inicial) & (player_game['season'] <= periodo_final)]
decada = pontuadores.nlargest(10, 'pts_per_game')[['player','pos','age', 'season', 'experience','tm','pts_per_game']]
media_decada = decada["pts_per_game"].mean()

print('A média da década', periodo_inicial , 'foi de: ', media_decada.round(2))
print('Os maiores pontuadores da década ', periodo_inicial )
print(decada)

A média da década 1980 foi de:  32.93
Os maiores pontuadores da década  1980
                  player pos   age  season  experience   tm  pts_per_game
20453     Michael Jordan  SG  23.0    1987           3  CHI          37.1
20044     Michael Jordan  SG  24.0    1988           4  CHI          35.0
19154     Michael Jordan  SG  26.0    1990           6  CHI          33.6
22902      George Gervin  SG  27.0    1980           8  SAS          33.1
20970       Bernard King  SF  28.0    1985           8  NYK          32.9
19619     Michael Jordan  SG  25.0    1989           5  CHI          32.5
22170      George Gervin  SG  29.0    1982          10  SAS          32.3
22319       Moses Malone   C  26.0    1982           8  HOU          31.1
19090        Karl Malone  PF  26.0    1990           5  UTA          31.0
19891  Dominique Wilkins  SF  28.0    1988           6  ATL          30.7


# Década de 90

In [15]:
periodo_inicial = temp_90
periodo_final = temp_2000

pontuadores = player_game[(player_game['season'] >= periodo_inicial) & (player_game['season'] <= periodo_final)]
decada = pontuadores.nlargest(10, 'pts_per_game')[['player','pos','age', 'season', 'experience','tm','pts_per_game']]
media_decada = decada["pts_per_game"].mean()

print('A média da década', periodo_inicial , 'foi de: ', media_decada.round(2))
print('Os maiores pontuadores da década ', periodo_inicial )
print(decada)

A média da década 1990 foi de:  30.82
Os maiores pontuadores da década  1990
                  player pos   age  season  experience   tm  pts_per_game
19154     Michael Jordan  SG  26.0    1990           6  CHI          33.6
17828     Michael Jordan  SG  29.0    1993           9  CHI          32.6
18718     Michael Jordan  SG  27.0    1991           7  CHI          31.5
19090        Karl Malone  PF  26.0    1990           5  UTA          31.0
16422     Michael Jordan  SG  32.0    1996          11  CHI          30.4
18266     Michael Jordan  SG  28.0    1992           8  CHI          30.1
17647  Dominique Wilkins  SF  33.0    1993          11  ATL          29.9
17156     David Robinson   C  28.0    1994           5  SAS          29.8
14354   Shaquille O'Neal   C  27.0    2000           8  LAL          29.7
15874     Michael Jordan  SG  33.0    1997          12  CHI          29.6


# Década de 2000

In [16]:
periodo_inicial = temp_2000
periodo_final = temp_2010

pontuadores = player_game[(player_game['season'] >= periodo_inicial) & (player_game['season'] <= periodo_final)]
decada = pontuadores.nlargest(10, 'pts_per_game')[['player','pos','age', 'season', 'experience','tm','pts_per_game']]
media_decada = decada["pts_per_game"].mean()

print('A média da década', periodo_inicial , 'foi de: ', media_decada.round(2))
print('Os maiores pontuadores da década ', periodo_inicial )
print(decada)

A média da década 2000 foi de:  31.81
Os maiores pontuadores da década  2000
              player pos   age  season  experience   tm  pts_per_game
10973    Kobe Bryant  SG  27.0    2006          10  LAL          35.4
10686  Allen Iverson  PG  30.0    2006          10  PHI          33.0
12858  Tracy McGrady  SG  23.0    2003           6  ORL          32.1
10444    Kobe Bryant  SG  28.0    2007          11  LAL          31.6
10984   LeBron James  SF  21.0    2006           3  CLE          31.4
12897  Allen Iverson  SG  26.0    2002           6  PHI          31.4
10171  Allen Iverson  PG  31.0    2007          11  PHI          31.2
13399  Allen Iverson  SG  25.0    2001           5  PHI          31.1
11248  Allen Iverson  PG  29.0    2005           9  PHI          30.7
9166     Dwyane Wade  SG  27.0    2009           6  MIA          30.2


# Década de 2010

In [17]:
periodo_inicial = temp_2010
periodo_final = temp_2020

pontuadores = player_game[(player_game['season'] >= periodo_inicial) & (player_game['season'] <= periodo_final)]
decada = pontuadores.nlargest(10, 'pts_per_game')[['player','pos','age', 'season', 'experience','tm','pts_per_game']]
media_decada = decada["pts_per_game"].mean()

print('A média da década', periodo_inicial , 'foi de: ', media_decada.round(2))
print('Os maiores pontuadores da década ', periodo_inicial )
print(decada)

A média da década 2010 foi de:  31.48
Os maiores pontuadores da década  2010
                 player pos   age  season  experience   tm  pts_per_game
3135       James Harden  PG  29.0    2019          10  HOU          36.1
2464       James Harden  SG  30.0    2020          11  HOU          34.3
6387       Kevin Durant  SF  25.0    2014           7  OKC          32.0
4710  Russell Westbrook  PG  28.0    2017           9  OKC          31.6
2250       Bradley Beal  SG  26.0    2020           8  WAS          30.5
3837       James Harden  SG  28.0    2018           9  HOU          30.4
5328      Stephen Curry  PG  27.0    2016           7  GSW          30.1
8720       Kevin Durant  SF  21.0    2010           3  OKC          30.1
2308     Damian Lillard  PG  29.0    2020           8  POR          30.0
8746       LeBron James  SF  25.0    2010           7  CLE          29.7


# Década de 2020 até 2023

In [18]:
periodo_inicial = temp_2020
periodo_final = temp_ultima

pontuadores = player_game[(player_game['season'] >= periodo_inicial) & (player_game['season'] <= periodo_final)]
decada = pontuadores.nlargest(10, 'pts_per_game')[['player','pos','age', 'season', 'experience','tm','pts_per_game']]
media_decada = decada["pts_per_game"].mean()

print('A média da década', periodo_inicial , 'foi de: ', media_decada.round(2))
print('Os maiores pontuadores da década ', periodo_inicial )
print(decada)

A média da década 2020 foi de:  31.89
Os maiores pontuadores da década  2020
                       player pos   age  season  experience   tm  pts_per_game
2464             James Harden  SG  30.0    2020          11  HOU          34.3
306               Joel Embiid   C  28.0    2023           7  PHI          33.1
422               Luka Dončić  PG  23.0    2023           5  DAL          32.4
103            Damian Lillard  PG  32.0    2023          11  POR          32.2
2102            Stephen Curry  PG  32.0    2021          12  GSW          32.0
588   Shai Gilgeous-Alexander  PG  24.0    2023           5  OKC          31.4
1553             Bradley Beal  SG  27.0    2021           9  WAS          31.3
207     Giannis Antetokounmpo  PF  28.0    2023          10  MIL          31.1
1065              Joel Embiid   C  27.0    2022           6  PHI          30.6
2250             Bradley Beal  SG  26.0    2020           8  WAS          30.5


# Média 10 maiores pontuadores de todas temporadas

In [74]:
dez_maiores = player_game.nlargest(10, 'pts_per_game')
dez_maiores = dez_maiores["pts_per_game"].mean()
print('A média de pontos dos 10 maiores pontuadores de todas temporadas: ',primeira_temporada, 'até' ,ultima_temporada, 'foi de:' , dez_maiores.round(2))


A média de pontos dos 10 maiores pontuadores de todas temporadas:  1947 até 2023 foi de: 39.41


...
#  1) Jogadores com mais de 35 pontos por partida
#  1) Players with more than 35 points per match

In [48]:
#Jogadores com mais de 35 pontos por partida
maiores_pontuadores = player_game.query('pts_per_game > 35')[['player','season','pos','age', 'experience','tm','pts_per_game']]
maiores_pontuadores.sort_values('pts_per_game', ascending=False)


player  season pos   age  experience   tm  pts_per_game
28813  Wilt Chamberlain    1962   C  25.0           3  PHW          50.4
28679  Wilt Chamberlain    1963   C  26.0           4  SFW          44.8
28413  Wilt Chamberlain    1965   C  28.0           6  SFW          38.9
28921  Wilt Chamberlain    1961   C  24.0           2  PHW          38.4
28734      Elgin Baylor    1962  SF  27.0           4  LAL          38.3
29044  Wilt Chamberlain    1960   C  23.0           1  PHW          37.6
20453    Michael Jordan    1987  SG  23.0           3  CHI          37.1
28549  Wilt Chamberlain    1964   C  27.0           5  SFW          36.9
3135       James Harden    2019  PG  29.0          10  HOU          36.1
28124        Rick Barry    1967  SF  22.0           2  SFW          35.6
10973       Kobe Bryant    2006  SG  27.0          10  LAL          35.4

________________________________________________________________________________________________________________________

# Análise jogador - Michael Jordan
# Player analysis - Michael Jordan

________________________________________________________________________________________________________________________


# 2) Michal Jordan - estatísticas por jogo no 1º ano de NBA
# 2) Michal Jordan - stats per game in 1st year of NBA
![Jordan](https://s2.glbimg.com/aH8Qm_BhkDtklFUpMzPPdKD1Fz3clJuUlwbRa-Y8b0SMKtSyFysUcLNrDB_hn4H8/s.glbimg.com/es/ge/f/original/2013/02/17/bas_michaeljordan01_get.jpg)


Idade: 21

ast_per_game: 5.9

stl_per_game: 2.4

blk_per_game: 0.8

pts_per_game: 28.2


Legendas

ast - assistência 

stl - roubo de bola

pts - pontos


Fonte: https://ge.globo.com/basquete/noticia/2013/02/jordan-50-motivos-para-ser-o-maior-jogador-de-todos-os-tempos.html

In [30]:
# This player's second year (Michael Jordan)
player_game.query("player == 'Michael Jordan' and experience == 2")


seas_id  season  player_id          player  birth_year pos   age  \
20822    10425    1986       2193  Michael Jordan         NaN  SG  22.0   

       experience   lg   tm  ...  ft_percent  orb_per_game  drb_per_game  \
20822           2  NBA  CHI  ...        0.84           1.3           2.3   

       trb_per_game  ast_per_game  stl_per_game  blk_per_game  tov_per_game  \
20822           3.6           2.9           2.1           1.2           2.5   

       pf_per_game  pts_per_game  
20822          2.6          22.7  

[1 rows x 35 columns]

# 3) Michal Jordan - estatísticas por jogo no 2º ano de NBA
# 3) Michal Jordan - stats per game in 2nd year of NBA

![Jordan](https://sneakernews.com/wp-content/uploads/2015/04/michael-jordan-changed-a-ball-boy-life.jpg)


Idade: 22

ast_per_game: 2.9

stl_per_game: 2.1

blk_per_game: 1.2

pts_per_game: 22.7


Legendas

ast - assistência 

stl - roubo de bola

pts - pontos


Fonte: https://sneakernews.com/2015/04/22/how-a-rookie-michael-jordan-changed-a-lucky-ball-boys-life/

# 4) Michal Jordan - estatísticas por jogo nos primeiros 5 anos de NBA
# 4) Michal Jordan - stats per game in the first 5 years of NBA
![Jordan](https://myhero.com/content/images/thumbs/0120508_michael-jordan.jpeg)


Fonte: https://myhero.com/content/images/thumbs/0120508_michael-jordan.jpeg

In [37]:
# 5 years statistics of each player (Michael Jordan)
player_game.query("player == 'Michael Jordan' and experience <= 5")[['player','age', 'experience','ast_per_game','stl_per_game','blk_per_game','pts_per_game']]


player   age  experience  ast_per_game  stl_per_game  \
19619  Michael Jordan  25.0           5           8.0           2.9   
20044  Michael Jordan  24.0           4           5.9           3.2   
20453  Michael Jordan  23.0           3           4.6           2.9   
20822  Michael Jordan  22.0           2           2.9           2.1   
21195  Michael Jordan  21.0           1           5.9           2.4   

       blk_per_game  pts_per_game  
19619           0.8          32.5  
20044           1.6          35.0  
20453           1.5          37.1  
20822           1.2          22.7  
21195           0.8          28.2

In [35]:
# This player's first year (Kobe Bryant)
player_game.query("player == 'Kobe Bryant' and experience == 1")

seas_id  season  player_id       player  birth_year pos   age  \
15809    15419    1997       3000  Kobe Bryant         NaN  SF  18.0   

       experience   lg   tm  ...  ft_percent  orb_per_game  drb_per_game  \
15809           1  NBA  LAL  ...       0.819           0.7           1.2   

       trb_per_game  ast_per_game  stl_per_game  blk_per_game  tov_per_game  \
15809           1.9           1.3           0.7           0.3           1.6   

       pf_per_game  pts_per_game  
15809          1.4           7.6  

[1 rows x 35 columns]

________________________________________________________________________________________________________________________

# Análise jogador - Kobe Bryant
# Player analysis - Kobe Bryant

________________________________________________________________________________________________________________________

# 2) Kobe Bryant - estatísticas por jogo no 1º ano de NBA
# 2) Kobe Bryant - stats per game in 1st year of NBA
![Kobe](https://bloximages.newyork1.vip.townnews.com/wenatcheeworld.com/content/tncms/assets/v3/editorial/5/ff/5ff86110-627a-5269-b3b3-a66d08afa4b8/647f5bfbaca4f.image.jpg)


Idade: 18

ast_per_game: 1.3

stl_per_game: 0.7

blk_per_game: 0.3

pts_per_game: 7.6


Legendas

ast - assistência 

stl - roubo de bola

pts - pontos


Fonte: https://bloximages.newyork1.vip.townnews.com/wenatcheeworld.com/content/tncms/assets/v3/editorial/5/ff/5ff86110-627a-5269-b3b3-a66d08afa4b8/647f5bfbaca4f.image.jpg




In [11]:
# This player's first year (Kobe Bryant)
player_game.query("player == 'Kobe Bryant' and experience == 2")

seas_id  season  player_id       player  birth_year pos   age  \
15919    15998    1998       3000  Kobe Bryant         NaN  SF  19.0   

       experience   lg   tm  ...  ft_percent  orb_per_game  drb_per_game  \
15919           2  NBA  LAL  ...       0.794           1.0           2.1   

       trb_per_game  ast_per_game  stl_per_game  blk_per_game  tov_per_game  \
15919           3.1           2.5           0.9           0.5           2.0   

       pf_per_game  pts_per_game  
15919          2.3          15.4  

[1 rows x 35 columns]

# 3) Kobe Bryant - estatísticas por jogo no 2º ano de NBA
![Kobe](https://cdn.vox-cdn.com/thumbor/vLjL0coUfwspg9W00xzT-6OTde8=/0x0:2398x3598/920x613/filters:focal(482x1135:864x1517):format(webp)/cdn.vox-cdn.com/uploads/chorus_image/image/66767315/75045057.jpg.0.jpg)


Idade: 19

ast_per_game: 2.3

stl_per_game: 0.9

blk_per_game: 0.5

pts_per_game: 15.4


Legendas

ast_per_game - assistência por jogo

stl_per_game - roubo de bola por jogo

blk_per_game - bloqueios por jogo

pts_per_game - pontos por jogo


Fonte: https://www.silverscreenandroll.com/2020/5/7/21250004/lakers-news-kobe-bryant-waves-off-karl-malone-1998-nba-all-star-game

# 4) Kobe Bryant - estatísticas por jogo nos primeiros 5 anos de NBA

![Kobe](https://papodesportivo.files.wordpress.com/2016/04/kobe.jpg?w=1024&h=640)


Fonte: https://papodesportivo.wordpress.com/2016/04/13/muito-obrigado-kobe-bryant/


In [13]:
# 5 years statistics of each player (Kobe Bryant)
player_game.query("player == 'Kobe Bryant' and experience <= 5")[['player','age', 'experience','ast_per_game','stl_per_game','blk_per_game','pts_per_game']]


player   age  experience  ast_per_game  stl_per_game  \
13693  Kobe Bryant  22.0           5           5.0           1.7   
14212  Kobe Bryant  21.0           4           4.9           1.6   
14720  Kobe Bryant  20.0           3           3.8           1.4   
15267  Kobe Bryant  19.0           2           2.5           0.9   
15809  Kobe Bryant  18.0           1           1.3           0.7   

       blk_per_game  pts_per_game  
13693           0.6          28.5  
14212           0.9          22.5  
14720           1.0          19.9  
15267           0.5          15.4  
15809           0.3           7.6

Alterando nome das colunas para ficar igual do dataset player_game



In [18]:
allstar.rename(columns={'AST':'ast_per_game','STL':'stl_per_game','BLK':'blk_per_game','PTS':'pts_per_game','Player':'player'}, implace=True)

TypeError: DataFrame.rename() got an unexpected keyword argument 'implace'

________________________________________________________________________________________________________________________

# Análise jogador - Lebron James
# Player analysis - Lebron James

________________________________________________________________________________________________________________________


# 2) Lebron James - estatísticas por jogo no 1º ano de NBA
![Lebron](https://a.espncdn.com/photo/2012/0112/nba_g_jamesleb_576.jpg)


Idade: 20

ast_per_game: 6.0

stl_per_game: 2.0

blk_per_game: 0.0

pts_per_game: 13


Legendas

ast - assistência 

stl - roubo de bola

pts - pontos


Fonte: https://a.espncdn.com/photo/2012/0112/nba_g_jamesleb_576.jpg


In [9]:
# This player's first year (Lebron James)
allstar.query("Player == 'Lebron James' and Age == 20")


Season  Age   Tm   Lg Pos    G   GS     MP   FG   FGA  ...  FT%  ORB  TRB  \
0  2004-05   20  CLE  NBA  SF  1.0  1.0  31:25  6.0  13.0  ...  0.0  1.0  8.0   

   AST  STL  BLK  TOV   PF   PTS        Player  
0  6.0  2.0  0.0  3.0  0.0  13.0  Lebron James  

[1 rows x 26 columns]

# 3) Lebron James - estatísticas por jogo no 2º ano de NBA
![Lebron](https://s3.amazonaws.com/content.espn.com.br/image/wide/622_47604d50-637b-3908-86dc-58f0ffb3f7b3.jpg)


Idade: 21

AST: 2.0

STL: 2.0

BLK: 0.0

PTS: 29


Legendas

AST - assistência 

STL - roubo de bola

BLK - bloqueios

PTS - pontos


Fonte: http://www.espn.com.br/noticia/493798_ha-10-anos-lebron-tinha-seu-primeiro-jogo-de-50-pontos-na-nba


In [73]:
# This player's second year (Lebron James)
allstar.query("Player == 'Lebron James' and Age == 21")


Season  Age   Tm   Lg Pos    G   GS     MP    FG   FGA  ...  FT%  ORB  \
1  2005-06   21  CLE  NBA  SF  1.0  1.0  30:34  12.0  21.0  ...  0.2  2.0   

   TRB  AST  STL  BLK  TOV   PF   PTS        Player  
1  6.0  2.0  2.0  0.0  1.0  2.0  29.0  Lebron James  

[1 rows x 26 columns]

# 6) Curiosidades / Dados relevantes

# Curiosities / Relevant data



# Conforme análise dos 10 maiores pontuadores por década (período disponível nesta base de dados 1947 a 1949)


### Primeiro ano de sua temporada já estando entre os 10 maiores pontuadores deste período.


###  No segundo ano de Liga, também foi um dos maiores pontuador deste período 

![George Mikan](https://jumperbrasil.lance.com.br/wp-content/uploads/2022/03/George-Mikan.jpg)



-------------- experience - 1 ------------------

age - 24

player -  30661

season - 1960

pos - C

tm  - Minneapolis Lakers   

pts_per_game - 28.3


-------------- experience - 2 ------------------

age - 25

player - 30379 

season - 1950

pos - C

tm  - Minneapolis Lakers  

pts_per_game - 27.4



Fonte: https://jumperbrasil.lance.com.br/lenda-nba-george-mikan-camisa-aposentada-lakers/


# Conforme análise dos 10 maiores pontuadores por década (período disponível nesta base de dados 1947 a 1949)


### Primeiro ano de sua temporada já estando entre os 10 maiores pontuadores deste período.


###  No segundo ano de Liga, também foi um dos maiores pontuador deste período 

![Ed Sadowski](https://celticswire.usatoday.com/wp-content/uploads/sites/39/2021/05/Ed-Macauley-Celtics-011552.jpg?resize=1000,600)


-------------- experience - 1 ------------------

age - 29

player -  30661

season - 1947

pos - C

tm  - Boston Celtics 

pts_per_game - 19.1



-------------- experience - 2 ------------------

age - 29

player -  30841

season - 1948

pos - C

tm  - Boston Celtics 

pts_per_game - 19.4


Fonte: https://celticswire.usatoday.com/tag/ed-sadowski/


# Conforme análise dos 10 maiores pontuadores por década 


### Primeiro ano de sua temporada, já estando entre os 10 maiores pontuadores deste período (1950 a 1959).

experience: 1

age: 23

player:  29044 

season: 1960

pos: C

tm  - Philadelphia 76 ers

pts_per_game - 37.6


###  No segundo ano de Liga, também foi um dos maiores pontuador deste período 

season: 1961, age: 24, experience: 2, tm:PHW,pts_per_game: 38.4



### Ocupa 4 posições de maior pontuador no período (1960 a 1959).

season: 1962, age: 25, experience: 3, tm:PHW,  pts_per_game:50.4

season: 1963, age: 26, experience: 4, tm:SFW, pts_per_game:44.8

season: 1965, age: 28, experience: 6, tm:SFW, pts_per_game: 38.9

season: 1961, age: 24, experience: 2, tm:PHW,pts_per_game: 38.4



###  Na lista de maiores pontuadores acima de 35 pontos de todos os tempos, ocupa a 1ª a 4ª posição

season: 1962, age: 25, experience: 3, tm:PHW,  pts_per_game:50.4

season: 1963, age: 26, experience: 4, tm:SFW, pts_per_game:44.8

season: 1965, age: 28, experience: 6, tm:SFW, pts_per_game: 38.9

season: 1961, age: 24, experience: 2, tm:PHW,pts_per_game: 38.4


![ Wilt Chamberlain](https://d1l5jyrrh5eluf.cloudfront.net/wp-content/uploads/2018/05/wilt.jpg)



Fonte: https://www.bing.com/images/search?view=detailV2&ccid=mg7T6wpp&id=3BE429401E3324F6000D6CD76885722AF8F5BDA7&thid=OIP.mg7T6wpp2DKeBLjX52foSQHaLH&mediaurl=https%3a%2f%2fd1l5jyrrh5eluf.cloudfront.net%2fwp-content%2fuploads%2f2018%2f05%2fwilt.jpg&cdnurl=https%3a%2f%2fth.bing.com%2fth%2fid%2fR.9a0ed3eb0a69d8329e04b8d7e767e849%3frik%3dp731%252bCpyhWjXbA%26pid%3dImgRaw%26r%3d0&exph=1028&expw=685&q=wilt+chamberlain&simid=607994952607036256&FORM=IRPRST&ck=038358BE4CA406CBADA5C10B8E8E7E41&selectedIndex=16&itb=0